### PR TITLE
Chores/user redirect logging #158949168

### PR DESCRIPTION
**Pull request by akegan**
Generates a sentry warning log if a user attempts to access a page in the application when they have not completed all previous sections. The log includes the fromState, toState, completedSections, and validatedForms

This will hopefully help us understand any looping issues in the future. 

image

image

### CODE REVIEW

**Comment by chrisdolendo** [comment]
@akegan just for future notice, we name PRs by their branch name and style them without dashes :)  Did this one for you. Will review shortly!

File: app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee [hunk 1]
```diff
@@ -604,7 +604,16 @@ ShortFormApplicationService = (
...
   Service.authorizedToProceed = (toState, fromState, toSection) ->
     return true unless toState && fromState
-    return false unless Service.userCanAccessSection(toSection.name)
+    if not Service.userCanAccessSection(toSection.name)
+      Raven.captureMessage('User attempted to access unauthorized section', {
+        level: 'warning',
+        extra: {
+          toState: toState.url, fromState: fromState,
+          completedSections: Service.application.completedSections,
+          validatedForms: Service.application.validatedForms
```
Comment (chrisdolendo) [inline app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee]: @akegan wondering if it's a good idea to log the whole application itself to see if the data looks legit (e.g. if a section is marked as incomplete and actually is from the values submitted etc).

Comment (akegan) [reply app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee]: is it OK to log people's personal info in sentry?

here's an example screenshot:
image

Comment (akegan) [reply app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee]: updated to include application details. i think we need to revisit in the future and potentially implement scrubbing for sensitive fields

**Pull request merged**